pkg/component/autoscaling/gardenercustommetrics/component.go

```diff
@@ -0,0 +1,215 @@
+	if err != nil {
+		return fmt.Errorf(baseErrorMessage+
+			" - failed to create the K8s object definitions which describe the individual "+
+			"k8s objects comprising the application deployment arrangement. "+
+			"The error message reported by the underlying operation follows: %w",
+			err)
+	}
+
+	err = managedresources.CreateForSeed(
+		ctx,
+		gcmx.seedClient,
+		gcmx.namespaceName,
+		managedResourceName,
+		false,
+		kubeObjects)
+	if err != nil {
+		return fmt.Errorf(baseErrorMessage+
+			" - failed to deploy the necessary resource config objects as a ManagedResource named '%s' to the server. "+
+			"The error message reported by the underlying operation follows: %w",
+			managedResourceName,
+			err)
+	}
+
+	return nil
+}
+
+// Destroy implements [component.Deployer.Destroy]()
+func (gcmx *GardenerCustomMetrics) Destroy(ctx context.Context) error {
+	if err := managedresources.DeleteForSeed(ctx, gcmx.seedClient, gcmx.namespaceName, managedResourceName); err != nil {
+		return fmt.Errorf(
+			"An error occurred while removing the gardener-custom-metrics component in namespace '%s' from the seed server"+
+				" - failed to remove ManagedResource '%s'. "+
+				"The error message reported by the underlying operation follows: %w",
+			gcmx.namespaceName,
+			managedResourceName,
+			err)
+	}
+
+	return nil
+}
+
+// Wait implements [component.Waiter.Wait]()
+func (gcmx *GardenerCustomMetrics) Wait(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, managedResourceTimeout)
+	defer cancel()
+
+	if err := managedresources.WaitUntilHealthy(timeoutCtx, gcmx.seedClient, gcmx.namespaceName, managedResourceName); err != nil {
+		return fmt.Errorf(
+			"An error occurred while waiting for the deployment process of the gardener-custom-metrics component to "+
+				"'%s' namespace in the seed server to finish and for the component to report ready status"+
+				" - the wait for ManagedResource '%s' to become healty failed. "+
+				"The error message reported by the underlying operation follows: %w",
+			gcmx.namespaceName,
+			managedResourceName,
+			err)
+	}
+
+	return nil
+}
+
+// WaitCleanup implements [component.Waiter.WaitCleanup]()
+func (gcmx *GardenerCustomMetrics) WaitCleanup(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, managedResourceTimeout)
+	defer cancel()
+
+	if err := managedresources.WaitUntilDeleted(timeoutCtx, gcmx.seedClient, gcmx.namespaceName, managedResourceName); err != nil {
+		return fmt.Errorf(
+			"An error occurred while waiting for the gardener-custom-metrics component to be fully removed from the "+
+				"'%s' namespace in the seed server"+
+				" - the wait for ManagedResource '%s' to be removed failed. "+
+				"The error message reported by the underlying operation follows: %w",
+			gcmx.namespaceName,
+			managedResourceName,
+			err)
+	}
+
+	return nil
```

Comment:
```suggestion
	return managedresources.WaitUntilDeleted(timeoutCtx, gcmx.seedClient, gcmx.namespaceName, managedResourceName)
```

Similar comment as above - pls strive for consistency compared to other component packages.